packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -401,6 +409,7 @@ export class Cluster extends Resource implements ICluster {
     // permissions and role are scoped. This will return `undefined` if kubectl
     // is not enabled for this cluster.
     this._k8sResourceHandler = this.createKubernetesResourceHandler();
+    this._helmChartHandler = this.createHelmChartHandler();
```

Comment:
Let's make this lazy: only create the handler in case a HelmChart resource is defined...

Reply:
What do you suggest, either expose a method on the Cluster construct `getOrCreateHelmChartHandler()` or make it part of the HelmChart construct itself?

Reply:
Moving it to the helm chart construct would actually be best, but it needs to be a singleton per cluster so you’ll want to attach it to the cluster scope and not per HelmChart of course (using node.tryFindChild as a condition)

Reply:
Added the `getOrCreateHelmChartHandler()` function in the HelmChart construct referencing the provided cluster to lazy add the handler.